### ISSUE TITLE
! fails for non-integers

### ISSUE THREAD

**Issue by Sp3000** [cheddar-lang/Cheddar#49]
**Version:** 0.3.0

### Problem

```
cheddar> !1
 false
cheddar> !0
 true
cheddar> !true
 `!` has no behavior for types `nil` and `Boolean`
cheddar> !false
 `!` has no behavior for types `nil` and `Boolean`
cheddar> !"abc"
 `!` has no behavior for types `nil` and `String`
```

### Desired Behavior

Proper outputs of `true`/`false`.

**Comment by vihanb** [comment]
Should empty strings be falsy? What about empty arrays?

**Comment by vihanb** [comment]
Fixed in 365d33df5362ad5c768902ac0d2924ff92dff8b2
